dataplane/saiserver/acl.go

```diff
@@ -292,3 +292,125 @@ func (a *acl) CreateAclEntry(ctx context.Context, req *saipb.CreateAclEntryReque
 
 	return &saipb.CreateAclEntryResponse{Oid: id}, nil
 }
+
+type myMacInfo struct {
+	priority       *uint32
+	portID         *uint64 // wildcard if not specified
+	vlanID         *uint32 // wildcard if not specified
+	macAddress     []byte
+	macAddressMask []byte
+}
+
+// ToEntryDesc returns the EntryDesc.
+func (mi *myMacInfo) ToEntryDesc(m *myMac, withPriority bool) (*fwdpb.EntryDesc, error) {
+	ed := &fwdpb.EntryDesc{
+		Entry: &fwdpb.EntryDesc_Flow{
+			Flow: &fwdpb.FlowEntryDesc{
+				Bank: 1,
+				Fields: []*fwdpb.PacketFieldMaskedBytes{{
+					FieldId: &fwdpb.PacketFieldId{Field: &fwdpb.PacketField{FieldNum: fwdpb.PacketFieldNum_PACKET_FIELD_NUM_ETHER_MAC_DST}},
+					Bytes:   mi.macAddress,
+					Masks:   mi.macAddressMask,
+				}},
+			},
+		},
+	}
+	if withPriority {
+		ed.GetFlow().Priority = *mi.priority
+	}
+
+	if mi.vlanID != nil {
+		ed.GetFlow().Fields = append(
+			ed.GetFlow().Fields,
+			&fwdpb.PacketFieldMaskedBytes{
+				FieldId: &fwdpb.PacketFieldId{Field: &fwdpb.PacketField{FieldNum: fwdpb.PacketFieldNum_PACKET_FIELD_NUM_VLAN_TAG}},
+				Bytes:   binary.BigEndian.AppendUint32(nil, *mi.vlanID),
```

Comment:
vlan tag is a 16 bit field, https://github.com/openconfig/lemming/blob/main/dataplane/forwarding/protocol/attr.go#L94